test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java

```diff
@@ -0,0 +1,294 @@
+
+        //try a real user and get an sso token back
+        String cookie = given().auth().basic("admin", "admin").when().get(url).then().log().all()
+                .assertThat().statusCode(200)
+                .assertThat().header("Set-Cookie", containsString("JSESSIONID")).extract().cookie("JSESSIONID");
+
+        //try the sso token instead of basic auth
+        given().cookie("JSESSIONID", cookie).when().get(url).then().log().all().assertThat().statusCode(200);
+
+        //try that admin level sso token on a restricted resource and get in... sso works!
+        given().cookie("JSESSIONID", cookie).when().get("https://localhost:9993/admin/index.html").then().log().all().assertThat().statusCode(200);
+    }
+
+    private void configureRestForAnonymous() throws IOException, InterruptedException {
+        PolicyProperties policyProperties = new PolicyProperties();
+        policyProperties.put("authenticationTypes", "/=SAML|ANON,/admin=SAML|basic,/jolokia=SAML|basic,/system=SAML|basic,/solr=SAML|PKI|basic");
+        policyProperties.put("whiteListContexts", "/services/SecurityTokenService,/services/internal,/proxy," + SERVICE_ROOT + "/sdk/SoapService");
+        Configuration config = configAdmin.getConfiguration(PolicyProperties.FACTORY_PID, null);
+        Dictionary<String, ?> configProps = new Hashtable<>(policyProperties);
+        config.update(configProps);
+        waitForAllBundles();
+    }
+
+    private void configureRestForBasic() throws IOException, InterruptedException {
+        PolicyProperties policyProperties = new PolicyProperties();
+        policyProperties.put("authenticationTypes", "/=SAML|basic,/admin=SAML|basic,/jolokia=SAML|basic,/system=SAML|basic,/solr=SAML|PKI|basic");
+        policyProperties.put("whiteListContexts", "/services/SecurityTokenService,/services/internal,/proxy," + SERVICE_ROOT + "/sdk/SoapService");
+        Configuration config = configAdmin.getConfiguration(PolicyProperties.FACTORY_PID, null);
+        Dictionary<String, ?> configProps = new Hashtable<>(policyProperties);
+        config.update(configProps);
+        waitForAllBundles();
+    }
+
+    @Test
+    public void testAnonymousSoapAccess() throws Exception {
+        configureRestForAnonymous();
+
+        String body = "<soapenv:Envelope xmlns:hel=\"http://ddf.sdk/soap/hello\" xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                + "   <soapenv:Header>\n" + "   </soapenv:Header>\n" + "   <soapenv:Body>\n" + "      <hel:helloWorld/>\n" + "   </soapenv:Body>\n"
+                + "</soapenv:Envelope>";
+        //we are only testing anonymous because that hits the most code, testing with an assertion would be mostly testing the same stuff that this is hitting
+        given().log().all().body(body).header("Content-Type", "text/xml; charset=utf-8").header("SOAPAction", "helloWorld").expect().statusCode(200)
```

Comment:
I would suggest removing .expect().statusCode(200) and instead add .statusCode(equalTo(200)) after .assertThat() to get better assertion failure messages.